protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java

```diff
@@ -0,0 +1,254 @@
+	   System.out.println("XmpDetector created, but null LOG");
+	}
+	else {
+	    LOG.debug("XmpDetector created, service "+xmpServiceName+" at "+createTimeDate);
+	}
+
+    } /* XmpDetector */
+
+    public XmpDetector()
+    {
+	this(DEFAULT_SERVICE_NAME,Xmp.XMP_PORT);
+        
+    } /* XmpDetector */
+
+    public String getServiceName() {
+	LOG.debug("XmpDetector: getServiceName");
+	return xmpServiceName;
+    }
+  
+    public void setServiceName(String newServiceName) 
+    {
+	LOG.debug("XmpDetector: setServiceName to "+newServiceName);
+    	xmpServiceName = new String(newServiceName);         
+    }
+
+    public void init() { onInit(); }
+
+    public void onInit()
+    {
+        LOG.debug("XmpDetector: onInit starting");
+	
+        // try to get configuration
+        try { 
+            XmpConfig protoConfig;
+
+	    XmpConfigFactory.init();
+            protoConfig = XmpConfigFactory.getInstance().getXmpConfig();
+            if (protoConfig.hasPort())
+               xmpPort = protoConfig.getPort();
+            if (protoConfig.hasTimeout())
+               xmpTimeout = protoConfig.getTimeout();
+            if (protoConfig.getAuthenUser() != null)
+               xmpAuthenUser = protoConfig.getAuthenUser();
+
+            sockopts.setConnectTimeout(xmpTimeout);	    
+
+        } catch (Throwable e) {
+            if (LOG != null)
+               LOG.error("XmpDetector: no config factory, using defaults");   
+            else
+	       System.out.println("XmpDetector: null LOG");
+        }
+    }
+
+    public int getPort()
+    {
+        LOG.debug("XmpDetector: getPort");
+	return xmpPort;
+    }
+
+    public void setPort(int newPort)
+    {
+        LOG.debug("XmpDetector: setPort to "+newPort);
+	xmpPort = newPort;
+    }
+
+    public void setIpMatch(String ipMatch)
+    {
+	LOG.debug("XmpDetector: setIpMatch to "+ipMatch);
+	
+	m_ipMatch = ipMatch;
+    }
+
+    public String getIpMatch()
+    {
+        LOG.debug("XmpDetector: getIpMatch returning '"+m_ipMatch+"'");
+	return m_ipMatch;
+    }
+
+    public int getTimeout()
+    {
+        LOG.debug("XmpDetector: getTimeout returning "+xmpTimeout);
+	return xmpTimeout;
+    }
+
+    public void setTimeout(int newTimeout) 
+    {
+        LOG.debug("XmpDetector: setTimeout to "+newTimeout);
+        xmpTimeout = newTimeout; 
+        sockopts.setConnectTimeout(xmpTimeout);
+    }
+
+    public void dispose()
+    {
+	LOG.debug("XmpDetector: dispose invoked");
+	
+	// dispose of anything like sessions, etc.
+        // no need to dispose SocketOpts
+    }
+
+    @Override
+    public final boolean isServiceDetected(InetAddress address)
+    {
+        XmpSession aSession;
+        XmpMessage aReply;
+        XmpVar[] vars,replyVars;
+
+	LOG.debug("XmpDetector: isServiceDetected starting to query "+address);
+	
+        // try to establish session
+        aSession = new XmpSession(sockopts,address,xmpPort,xmpAuthenUser);
+        if (aSession == null) {
```

Comment:
True.  Thanks.  However, I still have to make the point that given some of the other ONMS code I've reviewed, I'm struggling to understand why you would hold up this submission over a extraneous if statement.